src/main/java/frc/robot/utilities/pid/TrigonPIDFTalonFX.java

```diff
@@ -0,0 +1,109 @@
+package frc.robot.utilities.pid;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.DemandType;
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+import frc.robot.components.TrigonTalonSRX;
+import frc.robot.utilities.CTREUtil;
+import frc.robot.utilities.MotorConfig;
+
+public class TrigonPIDFTalonFX extends TrigonTalonSRX implements PIDFMotor {
+    private PIDFCoefs pidfCoefs;
+    private int pidSlotId;
+    private boolean isTuningPID;
+    private double tuningPIDSetpoint;
+
+    /**
+     * constructs a new PID motor controller
+     *
+     * @param id          device ID of motor controller
+     * @param motorConfig The configuration preset to use
+     * @param pidfCoefs   the pidf coefficients to use
+     * @param pidSlotId   the pid slot ID to use
+     */
+    public TrigonPIDFTalonFX(int id, MotorConfig motorConfig, PIDFCoefs pidfCoefs, int pidSlotId) {
```

Comment:
so many constructors... let the user put in default values himself?
+ why not include the coefs in `MotorConfig`

Reply:
1. you will many many times want to create a motorconfig without a coefs. 
2. motorConfig is a bit too crowded.